### ISSUE TITLE
diff between code and paper

### ISSUE THREAD

**Issue by jiangh0** [hkust-nlp/llm-compression-intelligence#5]
![image](https://github.com/hkust-nlp/llm-compression-intelligence/assets/24559732/0c1e36a4-52e2-49d6-a388-ef0618d1aa2f)
![image](https://github.com/hkust-nlp/llm-compression-intelligence/assets/24559732/0d4dfb28-eafd-43b8-b5fe-5d177740ba3a)
T is the character length in paper. But there is `np.log(2)` in code ?

**Comment by HYZ17** [comment]
As we mentioned in the footnote of our paper "Libraries like PyTorch often implement the cross entropy loss with the log base of e rather than 2, which causes a constant coefficient difference. ", we need to multiple by log2 to calculate BPC.

**Comment by jiangh0** [comment]
What the version of `torch`, `transformers` do you use?
And how long does it take for the 120b model to compute bpc?

**Comment by HYZ17** [comment]
We used transformers == 4.40.0, pytorch == 2.0.1. Since we did not use version-specific features, the latest version should work well.  FYI, Llama-3-70B took about 1 day on 6 A100 GPUs to complete testing in all three abilities. (With flash attention 2.0)

**Comment by jiangh0** [comment]
I find that the num of examples is different in different models.
Such as `cc`, the amount of examples is 7396 in qwen7b and that is 8489 in llama2-7b.

![image](https://github.com/hkust-nlp/llm-compression-intelligence/assets/24559732/53dd43a9-29d0-4b92-9107-a27bac893774)
![image](https://github.com/hkust-nlp/llm-compression-intelligence/assets/24559732/a87d42fe-72f2-4f01-802d-0b4e0605febf)

**Comment by HYZ17** [comment]
The number of iteration is determined by tokenizer and batch size. Since the tokenizers of these two model are different, it should be fine.

**Comment by jiangh0** [comment]
I assess all models with our own evaluating codes. But I find there is much difference. 
I find that my results have a bigger variance than you. And the pearson correlation coefficient is not very high.
I wanna to know how you assess these models. 

1. Can you open source the evaluating codes?
2. What is the prompt used in the evaluating? How many shot do you used in evaluating?
3. Are you evaluating all the data in the dataset？

**Comment by jiangh0** [comment]
![image](https://github.com/hkust-nlp/llm-compression-intelligence/assets/24559732/ea45c8c2-7bef-4f32-a9db-05264c312b56)
The result of `llama2-70b` and `deepseek-llm-67b` of `MMLU` is different between your paper and deepseek. https://github.com/deepseek-ai/DeepSeek-LLM. They also use ppl and 5-shot to evaluate the model. Is there any difference causing the results to be inconsistent

**Comment by HYZ17** [comment]
As mentioned in our paper, 

> for MMLU evaluation, we employ a cloze-style input format for this popular multiple-choice question (MCQ) benchmark, presenting only the question without the options, and subsequently selecting the choice with the highest likelihood. 

The cloze-style results is much different from the default style you mentioned. The comparison between this two style is in the Appendix D. The results are shown in the figure below. The results of llama2-70b and deepseek-llm-67b in default-style are actually close to the official results.
<img width="595" alt="image" src="https://github.com/hkust-nlp/llm-compression-intelligence/assets/73088368/609861b6-e72a-4f50-add1-c52d0bf55b0d">

**Comment by jiangh0** [comment]
Thank you for all your answers. 
I would like to ask, except the MMLU, are other datasets also evaluated in the MCQ style? 
What code repository is the MMLU cloze-style evaluation based on, and could you provide the dataset and evaluation repository? 
And What is the reason for evaluating multiple-choice question datasets using perplexity-based method?

**Comment by jiangh0** [comment]
And what is the reason of choosing 25-shot for `ARC-Challenge` and 0-shot for `Hellaswag`? This doesn't seem to be a universal setting.

**Comment by HYZ17** [comment]
1. please refer to our paper for details.
2. FYI, we modified a few lines of code from [lm-evaluation-harness](https://github.com/EleutherAI/lm-evaluation-harness/tree/b281b0921b636bc36ad05c0b0b0763bd6dd43463) to implement cloze-style MMLU.
3. ppl-based method is a commonly used in MCQ tasks evaluation. Plz refer to the tech report of llama1, llama2, deepseek,  qwen...
4. No special reason. Both zero-shot and few-shot are commonly used for these knowledge benchmark. For example, GPT -4 uses 25-shot for ARC-challenge evaluation and 10-shot for Hellaswag. And llama-2 used zero-shot for all these benchmarks. As long as all models use the same evaluation method, the results are comparable

**Comment by Kira0096** [comment]
> 1. please refer to our paper for details.
> 2. FYI, we modified a few lines of code from [lm-evaluation-harness](https://github.com/EleutherAI/lm-evaluation-harness/tree/b281b0921b636bc36ad05c0b0b0763bd6dd43463) to implement cloze-style MMLU.
> 3. ppl-based method is a commonly used in MCQ tasks evaluation. Plz refer to the tech report of llama1, llama2, deepseek,  qwen...
> 4. No special reason. Both zero-shot and few-shot are commonly used for these knowledge benchmark. For example, GPT -4 uses 25-shot for ARC-challenge evaluation and 10-shot for Hellaswag. And llama-2 used zero-shot for all these benchmarks. As long as all models use the same evaluation method, the results are comparable

For evaluating cloze form MMLU, how many shot do you choose?

**Comment by jiangh0** [comment]
I read the code of `LM-Evaluation-Harness` and I find that the evaluation method of this repository are all based on `ppl`.
And the paper says that `default style` `MMLU` is evaluated with `LM-Evaluation-Harness`.
So is  the result of `default style` `MMLU` evaluated on ppl method too? with `MCQ` prompt? just calculate ppl of A/B/C/D to select the answer?

**Comment by HYZ17** [comment]
Yes.

**Comment by jiangh0** [comment]
![image](https://github.com/hkust-nlp/llm-compression-intelligence/assets/24559732/1c9deeb3-2672-4bad-9566-d1521626c219)
![image](https://github.com/hkust-nlp/llm-compression-intelligence/assets/24559732/f7385a0e-1eba-4e8f-a57f-ea1f0e4e38ab)
For `cloze-style` and `defalut-style` `MMLU`, I constructed the prompt in above pictures based on my own ideas.
I wanna to know is there any difference from your prompt?
Is there the content of choices in `Answer` of `cloze-style MMLU`?

**Comment by HYZ17** [comment]
For the cloze-style, yes, ours are the same. For the default style, I believe in lm-evaluation-harness, the prompts are in the  format as follows.

> Find all c in Z_3 such that Z_3[x]/(x^2 + c) is a field.
> A. 0
> B. 1
> C. 2
> D. 3
> Answer: B
>
